### PR TITLE
PSEA

### PR DESCRIPTION
**Pull request by domdinicola**

### CODE REVIEW

**Comment by domdinicola** [comment]
@reinbach I don't think is a good idea to call the model Engagement... we already have another class in `audit` app called Engagement, would be better to find another name in order to avoid confusion.

**Comment by reinbach** [comment]
@domdinicola How about `Assessment`?

**Comment by domdinicola** [comment]
@reinbach we have Assessment in `partner`..

**Comment by reinbach** [comment]
@domdinicola I'd say it not really a big deal, the app encapsulates it fine.

File: src/etools/applications/psea/models.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    def rating(self):
+        result = Answer.objects.filter(assessment=self).aggregate(
+            assessment=Sum("rating__weight")
+        )
+        if result:
+            return result["assessment"]
+        return None
+
+    def get_reference_number(self):
```
Comment (robertavram) [inline src/etools/applications/psea/models.py]: ref number can be "{}/{}PSEA{}".format(country_short_code, current_year, id of record)

File: src/etools/applications/psea/models.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+    class Meta:
+        verbose_name = _('Evidence')
+        verbose_name_plural = _('Evidence')
+
+    def __str__(self):
+        return f'{self.label}'
+
+
+class Indicator(TimeStampedModel):
```
Comment (robertavram) [inline src/etools/applications/psea/models.py]: I think we'll have to come up with some better naming for some of these models.

File: src/etools/applications/psea/models.py (hunk 1 above)
Comment (reinbach) [reply src/etools/applications/psea/models.py]: :+1:

File: src/etools/applications/psea/models.py (hunk 2 above)
Comment (reinbach) [reply src/etools/applications/psea/models.py]: They seem appropriate/applicable to me

File: src/etools/applications/psea/serializers.py [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+
+class BaseAssessmentSerializer(serializers.ModelSerializer):
+    permissions = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = Assessment
+
+    def get_permissions(self, obj):
+        user = self.context['request'].user
+        if self.context["view"].action == "list":
```
Comment (robertavram) [inline src/etools/applications/psea/serializers.py]: I think there is some confusion here.. we shouldn't serialize permissions for the users in List.. not sure why we're doing this.. One thing we need is to filter out all assessments that user does not belong in if the user is not unicef.
I'll make the change

**Pull request merged**